lib/services/local/configuration.go

```diff
@@ -218,7 +218,7 @@ func (s *ClusterConfigurationService) DeleteAuthPreference(ctx context.Context)
 	return nil
 }
 
-// GetClusterConfig gets services.ClusterConfig from the backend.
+// GetClusterConfig gets types.ClusterConfig from the backend.
 func (s *ClusterConfigurationService) GetClusterConfig(opts ...services.MarshalOption) (types.ClusterConfig, error) {
```

Comment:
This method should be updated to not fail when the `ClusterConfig` object is not in the backend.
Instead, it should generate a valid legacy `types.ClusterConfig` from all the new config objects.

This will keep backwards-compatibility for a case of:
- new cluster, with auth server running v7.x
- v6.x nodes joining this new cluster